pkg/cached/cache.go

```diff
@@ -187,7 +191,70 @@ func (c *merger[K, T, V]) Get() Result[V] {
 	return c.result
 }
 
-type transformerCacheKeyType struct{}
+// NewListMerger creates a new merge cache that merges the results of
+// other caches in list form. The function only gets called if any of
+// the dependency has changed.
+//
+// The benefit of ListMerger over the basic Merger is that caches are
+// stored in an ordered list so the order of the cache will be
+// preserved in the order of the results passed to the mergeFn.
+//
+// If any of the dependency returned an error before, or any of the
+// dependency returned an error this time, or if the mergeFn failed
+// before, then the function is reran.
+//
+// Note that this assumes there is no "partial" merge, the merge
+// function will remerge all the dependencies together everytime. Since
+// the list of dependencies is constant, there is no way to save some
+// partial merge information either.
```

Comment:
Sorry, I just realized you've put the doc on the private type, not on the public constructor.